test/unit/host_discovered_test.rb

```diff
@@ -380,39 +380,46 @@ class HostDiscoveredTest < ActiveSupport::TestCase
 
   test "primary interface isn't touched with no LLDP facts" do
     Setting[:discovery_auto_bond] = true
-    raw = parse_json_fixture('/facts.json')['facts']
+    raw = parse_json_fixture('regular_host', true)
     host = discover_host_from_facts(raw)
     refute_nil host.primary_interface
     assert_equal "eth0", host.primary_interface.identifier
   end
 
+  test "primary interface via tagged VLAN is discovered with an IP" do
+    raw = parse_json_fixture('rhel7-vlan')
+    host = discover_host_from_facts(raw)
+    assert_equal "192.168.1.16", host.ip
```

Comment:
shouldn't the line above have `host.primary_interface.ip`?

Reply:
I should have been more verbose - this test will fail until the core PR is merged as stated in the description. Sorry, I will remove this from this PR and move it somewhere else.